src/XR/motionController/webXRGenericHandController.ts

```diff
@@ -0,0 +1,138 @@
+import { IMotionControllerLayoutMap, IMinimalMotionControllerObject, MotionControllerHandedness, WebXRAbstractMotionController } from "./webXRAbstractMotionController";
+import { Scene } from "../../scene";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Mesh } from "../../Meshes/mesh";
+import { Quaternion } from "../../Maths/math.vector";
+import { WebXRMotionControllerManager } from "./webXRMotionControllerManager";
+
+/**
+ * The motion controller class for the standard HTC-Vive controllers
+ */
+export class WebXRGenericHandController extends WebXRAbstractMotionController {
+
+    public profileId = "generic-hand-select-grasp";
+
+    /**
+     * Create a new Vive motion controller object
+     * @param scene the scene to use to create this controller
+     * @param gamepadObject the corresponding gamepad object
+     * @param handedness the handedness of the controller
+     */
+    constructor(scene: Scene, gamepadObject: IMinimalMotionControllerObject, handedness: MotionControllerHandedness) {
+        // Don't load the controller model - for now, hands have no real model.
+        super(scene, GenericHandSelectGraspProfile[handedness], gamepadObject, handedness, true);
+    }
+
+    protected _getFilenameAndPath(): { filename: string; path: string } {
+        return {
+            filename: "generic.babylon",
+            path: "https://controllers.babylonjs.com/generic/",
+        };
+    }
+
+    protected _getModelLoadingConstraints(): boolean {
+        return true;
+    }
+
+    protected _processLoadedModel(_meshes: AbstractMesh[]): void {
+        // nothing to do
+    }
+
+    protected _setRootMesh(meshes: AbstractMesh[]): void {
+        this.rootMesh = new Mesh(this.profileId + " " + this.handedness, this.scene);
+
+        meshes.forEach((mesh) => {

```

Comment:
probably not needed

Reply:
> LGTM!
> Just one question - this profile exists in the webxr-input repository, which we load per default. Won't it load this profile automatically from the repository if the profile is set in the XRInputSource?

It looks like the motion controller manager's UpdateProfilesList() pulls from the list found here:
https://immersive-web.github.io/webxr-input-profiles/packages/viewer/dist/profiles/profilesList.json
But this doesn't hold any data for hands, since it looks like a list of controller models.